pkg/s3/utils.go

```diff
@@ -12,15 +14,19 @@ import (
 
 var isEC2Instance bool
 var isEC2InstanceOnce sync.Once
+var isEC2InstanceTimeout = 2 * time.Second
 
 func DefaultAWSConfig() (aws.Config, error) {
-	imdsMode := imds.ClientEnabled
-	if !IsEC2Instance(context.Background()) {
-		imdsMode = imds.ClientDisabled
+	// Set a default IMDC TTL of 1 hour if not set to avoid hitting the metadata service too often, which can slow down the
+	// node's startup time.
+	if _, ok := os.LookupEnv("AWS_EC2_METADATA_TTL"); !ok {
+		err := os.Setenv("AWS_EC2_METADATA_TTL", "3600")
```

Comment:
might also want this to be configurable.

Reply:
It is configurable is you set `AWS_EC2_METADATA_TTL` :)
Besides, this method returns `DefaultAWSConfig` which should return default values that we believe are good in most cases. 
In general, I do agree that we have a lot of work to be done to make our settings more configurable. For now, all new types are accepting dependencies and configurations instead of instantiating them in the constructors. Most of the technical debt is accumulating in the `node` package

Reply:
🤦 right, sorry.